test/functionalTest/cases/2631_partial_datetimes/partial_datetime_creation.test

```diff
@@ -0,0 +1,226 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Support partial date time creations
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Creating entity with several attributes using all partial date variants, all meaning same date
+# 02. Get entity
+#
+
+echo "01. Creating entity with several attributes using all partial date variants, all meaning same date"
+echo "=================================================================================================="
+payload='{
+  "id": "E",
+  "type": "T",
+  "D01": {
+    "value": "2016-10-01",
+    "type": "DateTime"
+  },
+  "D02": {
+    "value": "2016-10-01Z",
```

Comment:
I understand that the concern if is "2016-10-01Z", "2016-10-01+01:00", etc. are legal or if they should use also the time part.

ISO8601 says: "If a time zone designator is required, it follows the combined date and time". My interpretation is that "2016-10-01Z" is *not* correct.

Morevoer, taking into account from https://en.wikipedia.org/wiki/ISO_8601#Times that the minimal time expression is `hh` (i.e. no "null" time expressions are allowed) them "2016-10-01TZ", "2016-10-01T+01:00" are also incorrect.

Thus, the "minimal" date expression with timezone is "2016-10-01T00Z", "2016-10-01T00+01:00".

Related with this (but applies to the whole implementation), ISO8601 also says "It is permitted to omit the 'T' character by mutual agreement". However, given that in general Orion and the client are unknown to each other (worst case assumption) I think we cannot omit the T.